runtime/browser/ui/desktop/xwalk_popup_controller.h

```diff
@@ -0,0 +1,111 @@
+// Copyright 2014 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef XWALK_RUNTIME_BROWSER_UI_DESKTOP_XWALK_POPUP_CONTROLLER_H_
+#define XWALK_RUNTIME_BROWSER_UI_DESKTOP_XWALK_POPUP_CONTROLLER_H_
+
+#include <utility>
+
+#include "base/i18n/rtl.h"
+#include "content/public/browser/render_widget_host.h"
+#include "ui/gfx/geometry/rect.h"
+#include "ui/gfx/geometry/rect_f.h"
+#include "ui/gfx/native_widget_types.h"
+
+namespace content {
+struct NativeWebKeyboardEvent;
+class RenderViewHost;
+class WebContents;
+}
+
+namespace gfx {
+class Display;
+}
+
+namespace xwalk {
+
+// Class that controls common functionality for Autofill style popups. Can
+// determine the correct location of a popup of a desired size and can register
+// a handler to key press events.
+class XWalkPopupController {
+ public:
+  XWalkPopupController(const gfx::RectF& element_bounds,
+      base::i18n::TextDirection text_direction,
+      gfx::NativeView container_view,
+      content::WebContents* web_contents);
+  virtual ~XWalkPopupController();
+
+  const gfx::RectF& element_bounds() const { return element_bounds_; }
+  bool is_rtl() const { return text_direction_ == base::i18n::RIGHT_TO_LEFT; }
+  gfx::NativeView container_view() { return container_view_; }
+  content::WebContents* web_contents() { return web_contents_; }
+
+  // Returns the enclosing rectangle for |element_bounds_|.
+  const gfx::Rect RoundedElementBounds() const;
```

Comment:
no sense in returning a const by value,
`gfx::Rect GetRoundedElementBounds() const;`